mono/metadata/threads.c

```diff
@@ -2671,8 +2671,10 @@ ves_icall_System_Threading_Thread_GetAbortExceptionState (MonoThreadObjectHandle
 	MonoExceptionHandle invalid_op_exc = mono_exception_new_invalid_operation (
 		"Thread.ExceptionState cannot access an ExceptionState from a different AppDomain", error_creating_exception);
 	mono_error_assert_ok (error_creating_exception);
-	if (g_assert (!is_ok (error)))
+	if (g_assert (!is_ok (error))) {
 		MONO_HANDLE_SET (invalid_op_exc, inner_ex, mono_error_convert_to_exception_handle (error));
+		error_init_reuse (error);
+	}
 	return deserialized;
```

Comment:
This `return deserialized` is wrong.

The point is that on this path we need to set `error` to an invalid operation exception:

```c
mono_error_set_exception_instance (error, invalid_op_exc);
return NULL_HANDLE
```